03_Fiches_thematiques/Fiche_api.Rmd

```diff
@@ -0,0 +1,281 @@
+# Travailler avec des API
+
+<!-- ## Tâches concernées et recommandations -->
+
+<!-- Quelques détails sur la tâche dont il s'agit -->
+
+<!-- ::: {.recommandation data-latex=""} -->
+
+<!-- Dire en 4-5 lignes comment il est recommandé de procéder: -->
+
+<!-- * le ou les *package*(*s*) dont l'usage est recommandé; si on recommande plusieurs *packages*, expliquer comment choisir lequel (en fonction de la taille des données, du format...) -->
+<!-- * les *packages* dont l'usage est déconseillé;  -->
+<!-- * les autres points de méthode essentiels. -->
+
+<!-- ::: -->
+
+## Rappels des notions essentielles sur les API
+
+### Qu'est-ce qu'une API ?
+
+Une *Application Programming Interface* (ou **API**) est une interface applicative de programmation qui permet d'utiliser une application existante pour restituer des données. Il s'agit d'une façade clairement délimitée par laquelle un logiciel offre des services à d'autres logiciels (ou utilisateurs). L'objectif est de fournir une porte d'accès à une fonctionnalité en cachant les détails de la mise en œuvre. 
+
+Ce terme peut faire peur, mais il s'agit en fait simplement d'une façon de restituer des données. Plutôt que d'attaquer directement des bases de données, qui sont volumineuses et complexes, des outils ont été développés pour faire le travail intermédiaire et restituer la donnée sous forme de service.
+
+À l'Insee comme ailleurs, la connexion entre les bases de données pour les nouveaux projets tend à se réaliser via des API, afin de limiter le nombre de personnes ayant accès aux bases, mais aussi pour faciliter les accès avec des services sur-mesures pour les utilisateurs.
+
+### Mise à disposition des API
+
+Les API peuvent utiliser une interface utilisateur.  
+
+Les API Insee mise à disposition sur le [catalogue des API](https://api.insee.fr/catalogue/) en proposent. Cette interface, permet :
+
+- de s'inscrire aux différents services ;
+- de visualiser les différentes requêtes proposées par les services ;
+- de lancer l'API depuis cette plateforme ;
+- de documenter les API.
+
+Cependant, l'utilisation de cette interface ne sera pas utile très longtemps, car l'utilisateur va vite se rendre compte qu'une API s'utilise via un logiciel de traitement pour automatiser un processus ou réaliser du chargement de masse.
+
+
+Pour d'autres API, mises à disposition par la [BDM au format SDMX](https://www.insee.fr/fr/information/2862759), ce point d'entré n'existe pas. De la documentation est mise à disposition pour permettre à l'utilisateur de comprendre l'utilisation du service. L'utilisateur devra ensuite utiliser l'API directement via son url par un navigateur internet ou un logiciel adapté (R, java, python...).
```

Comment:
```suggestion
Pour d'autres API, mises à disposition par la [banque de données macroéconomiques (BDM) au format SDMX (*Statistical Data and Metadata Exchange*)](https://www.insee.fr/fr/information/2862759), ce point d'entrée n'existe pas. De la documentation est mise à disposition pour permettre à l'utilisateur de comprendre l'utilisation du service. L'utilisateur devra ensuite utiliser l'API directement via son url par un navigateur internet ou un logiciel adapté (`R`, `Python`, `Java`...).
```

Il serait bien de préciser un peu ce qu'est le SDMX pour les gens qui, comme moi, ne connaissent pas vraiment. Une boîte remarque, à compléter, peut faire l'affaire:

~~~markdown
::: remarque
Le format SDMX (*Statistical Data and Metadata Exchange*) est une initiative visant à promouvoir la diffusion et les échanges de données et de métadonnées statistiques de manière cohérente. 
:::
~~~